chain/epoch_manager/src/lib.rs

```diff
@@ -439,37 +439,40 @@ impl EpochManager {
         Ok(epoch_info.validators[seat.tenant as usize].clone())
     }
 
-    pub fn get_all_block_producer_info(
+    /// Returns seats with all block producers in current epoch, with indicator on whether they are slashed or not.
+    pub fn get_all_block_producer_seats(
         &mut self,
         epoch_id: &EpochId,
         last_known_block_hash: &CryptoHash,
-    ) -> Result<Vec<(ValidatorStake, bool)>, EpochError> {
+    ) -> Result<Vec<(Seat, ValidatorStake, bool)>, EpochError> {
         let slashed = self.get_slashed_validators(last_known_block_hash)?.clone();
         let epoch_info = self.get_epoch_info(epoch_id)?;
         let mut result = vec![];
-        let mut validators: HashSet<AccountId> = HashSet::default();
         for seat in epoch_info.block_producers.iter() {
             let validator_stake = epoch_info.validators[seat.tenant as usize].clone();
-            if !validators.contains(&validator_stake.account_id) {
-                let is_slashed = slashed.contains_key(&validator_stake.account_id);
-                validators.insert(validator_stake.account_id.clone());
-                result.push((validator_stake, is_slashed));
-            }
+            let is_slashed = slashed.contains_key(&validator_stake.account_id);
+            result.push((seat.clone(), validator_stake, is_slashed));
```

Comment:
I don't think you need to add `seat` here. We just need `Vec<(ValidatorStake, bool)>` to be returned.

Reply:
++